pybamm/models/submodels/particle/fickian/fickian_single_particle.py

```diff
@@ -37,12 +37,27 @@ def get_fundamental_variables(self):
                 c_s_xav, ["positive electrode"], broadcast_type="primary"
             )
 
-        N_s_xav = self._flux_law(c_s_xav)
+        variables = self._get_standard_concentration_variables(c_s, c_s_xav)
+
+        return variables
+
+    def get_coupled_variables(self, variables):
+
+        c_s_xav = variables[
+            "X-average " + self.domain.lower() + " particle concentration"
+        ]
+        T_k_av = variables["Average " + self.domain.lower() + " electrode temperature"]
+
+        if self.domain == "Negative":
+            D_s = self.param.D_n(c_s_xav, T_k_av)
+        elif self.domain == "Positive":
+            D_s = self.param.D_p(c_s_xav, T_k_av)
+
+        N_s_xav = D_s * self._flux_law(c_s_xav)
```

Comment:
minor point, but should D be in the flux law and then have `self._flux_law(c,D)` ?